src/features/patients/pages/PatientForm.tsx

```diff
@@ -53,6 +67,20 @@ export const PatientForm: FC = () => {
     patientSchema,
   } = usePatientSchemaQuery();
 
+  const jsonPatientSchemaValidator = () => {
+    if (patientSchema?.jsonSchema) {
+      return ajv.compile(patientSchema?.jsonSchema);
+    }
+    return () => false;
+  };
+
+  const jsonPatientSurveySchemaValidator = () => {
+    if (patientSurveySchema?.uiSchema) {
+      return ajv.compile(patientSurveySchema?.jsonSchema);
+    }
+    return () => false;
+  };
```

Comment:
could use refs or something as well, but this'll do for now.. bit annoying because if the schemas are not defined, then the compiled validation function gets all messy with memoizing the json schema form and `onSubmit` callback which will never be re-rendered because we stop it from re-rendering, so if you do it other ways it can be a race between fetching the schema and having a valid schema as a closure within the validation function and memoizing the json schema form..

Anyway- general tidy needed with the memoizing of the form..?